text/0000-local-loop-bindings.md

```diff
@@ -0,0 +1,218 @@
+- Feature Name: local_loop_bindings
+- Start Date: 2018-12-25
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+This will allow an extended syntax for `loop` to accept local variables, which change for each iteration to simplify more complicated loop constructs and avoid mutable state.
+
+
+# Motivation
```

Comment:
Good idea to look at existing code. I just tried to figure out some useful examples myself and only came up with rewriting factorial.

Reply:
Hehe ;) You might also want to compare and contrast with iterators. For example, are there cases where this would make code that is hard to formulate with iterators easier to write?

Reply:
I don't think, such a comparison is that useful. The main cases for using this syntax are not for iteration over sequences. Even if something is possible to formulate using iterators, using loops may be easier to understand, especially, when not knowing enough rust to understand iterators.